daml/Tests/Distribution/Auction.daml

```diff
@@ -17,72 +17,111 @@ import Marketplace.Distribution.Bidding.Model qualified as Bidding
+  issuer <- onboardCustomer providers "Issuer"
+  issuerAsset <- depositAsset providers issuer auctionAsset issuer.mainAccount.id
+  auctionOfferCid <- bank `submit` exerciseCmd distributorRoleCid Distributor.OfferAuctionService with customer = issuer.customer; ..
+  auctionServiceCid <- issuer.customer `submit` exerciseCmd auctionOfferCid Auction.Accept with tradingAccount = issuer.mainAccount; allocationAccount = issuer.auctionLockedAccount; receivableAccount = issuer.mainAccount
 
   -- Create an Auction request and approve it
   let
     auctionId = "MyAuction"
-    quotedAssetId = Id with signatories = fromList [ agent, issuer.customer ]; label = "USD"; version = 0
+    quotedAssetId = Id with signatories = fromList [ bank, issuer.customer ]; label = "USD"; version = 0
   createAuctionRequestCid <- issuer.customer `submit` exerciseCmd auctionServiceCid Auction.RequestCreateAuction with
-      asset = newDistributionAsset
-      assetDepositCid = issuer.assetDepositCid
+      asset = auctionAsset
+      assetDepositCid = issuerAsset
       floorPrice = 5.0
       ..
-  auctionCid <- agent `submit` exerciseCmd auctionServiceCid Auction.CreateAuction with ..
+  auctionCid <- bank `submit` exerciseCmd auctionServiceCid Auction.CreateAuction with ..
 
   -- Add our parties as bidders
-  bidRequests@[aliceBidRequestCid, bobBidRequestCid, charlieBidRequestCid, daveBidRequestCid] <- forA parties \party ->
-      agent `submit` exerciseCmd party.biddingServiceCid Bidding.RequestBid with issuer = issuer.customer; asset = newDistributionAsset; ..
+  aliceBidRequestCid    <- submit bank do exerciseCmd alice.biddingServiceCid Bidding.RequestBid with issuer = issuer.customer; asset = auctionAsset; auctionId; quotedAssetId
+  bobBidRequestCid      <- submit bank do exerciseCmd bob.biddingServiceCid Bidding.RequestBid with issuer = issuer.customer; asset = auctionAsset; auctionId; quotedAssetId
+  charlieBidRequestCid  <- submit bank do exerciseCmd charlie.biddingServiceCid Bidding.RequestBid with issuer = issuer.customer; asset = auctionAsset; auctionId; quotedAssetId
+  daveBidRequestCid     <- submit bank do exerciseCmd dave.biddingServiceCid Bidding.RequestBid with issuer = issuer.customer; asset = auctionAsset; auctionId; quotedAssetId
 
-  pure (parties, originalDeposits, issuer, agent, custodian, auctionCid, auctionServiceCid, bidRequests)
+  pure Fixture with ..
+
+submitBid : Customer -> ContractId Bidding.R -> ContractId AssetDeposit -> Decimal -> Decimal -> Script (ContractId Bidding.Bid)
+submitBid party bidRequestCid depositCid price quantity = do
+  let allowPublishing = False
+  time <- getTime
+  submit party.customer do exerciseCmd party.biddingServiceCid Bidding.SubmitBid with ..
 
 auctionAndBidsFullyAllocated : Script ()
 auctionAndBidsFullyAllocated = do
-  (parties, originalDeposits, issuer, auctionAgent, custodian, auctionCid, auctionServiceCid, bidRequests) <- setup
+  Fixture{..} <- setup
```

Comment:
Use explicit extraction here so e.g. `Fixture{issuer,auctionAgent}` instead of `..` to make explicit used variables (some of them don't appear to be used at all).